### PR TITLE
Stop native parameters defaults changing chart type after save or load

### LINKED ISSUE #33208: Native parameters defaults change chart type after save or load
### Describe the bug

After saving a native question with parameters and a "non-sensible" chart type, running the query again switches the chart type back to a table. This is inconsistent with how running the query again works when the native question isn't saved, where the chart type doesn't change.

The same thing happens when you load the page of the saved question: the chart type automatically switches to a table.

Loom demo:
https://www.loom.com/share/b3a3d3ea7d2647cc9e9bd82e9debc24c?sid=b7d6ee8b-5012-4619-82e7-f5cf07190e22

### To Reproduce

1. Create a native SQL question and input the query text: 
```select distinct category from products where {{category}} order by category```
2. Set the variable to be a field filter mapped to PRODUCTS.CATEGORY and with a default value of "Doohickey" and "Gadget".
3. Run the query, and change the chart type to "Number".
4. Run the query again, and observe the chart type stays as "Number".
5. Save the question
6. Run the query again, and observe the chart type has changed to "Table". This is the bug.

### Expected behavior

The chart type shouldn't change when the query runs again

### Logs

_No response_

### Information about your Metabase installation

```JSON
master
```

### Severity

low

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by calherries**
Fixes https://github.com/metabase/metabase/issues/33208

The explanation of the bug and fix is in [this comment](https://github.com/metabase/metabase/pull/33211#discussion_r1295148474).

### CODE REVIEW

File: frontend/src/metabase/query_builder/selectors.js [hunk 1]
```diff
@@ -306,6 +306,12 @@ export const getOriginalQuestion = createSelector(
   (metadata, card) => metadata && card && new Question(card, metadata),
 );
 
+export const getOriginalQuestionWithParameterValues = createSelector(
+  [getMetadata, getOriginalCard, getParameterValues],
+  (metadata, card, parameterValues) =>
+    metadata && card && new Question(card, metadata, parameterValues),
+);
```
Comment (calherries) [inline frontend/src/metabase/query_builder/selectors.js:313]: There is a `getOriginalQuestion` selector, but it returns a question without any `parameterValues`. We need the `parameterValues` to tell whether the question really has "dirtied" or not when the query results are returned [here](https://github.com/metabase/metabase/pull/33211/files#diff-222944c63a643f354fcf9e8fd93575fd6429baa8d763755500ab25cfc50842e5R183).

File: frontend/src/metabase/dashboard/actions/data-fetching.js [hunk 2]
```diff
@@ -238,14 +238,9 @@ export const fetchDashboard = createThunkAction(
...
         ? getParameterValues(getState())
-        : getParameterValuesByIdFromQueryParams(
-            parameters,
-            queryParams,
-            metadata,
-            {
-              forcefullyUnsetDefaultedParametersWithEmptyStringValue: true,
-            },
-          );
+        : getParameterValuesByIdFromQueryParams(parameters, queryParams, {
```
Comment (calherries) [inline frontend/src/metabase/dashboard/actions/data-fetching.js:241]: `getParameterValuesByIdFromQueryParams` didn't use its `metadata` arg, so I removed it.

File: frontend/src/metabase/parameters/utils/parameter-values.js [hunk 3]
```diff
@@ -1,11 +1,7 @@
 import { getParameterType } from "metabase-lib/parameters/utils/parameter-type";
 import { hasParameterValue } from "metabase-lib/parameters/utils/parameter-values";
 
-export function getParameterValueFromQueryParams(
-  parameter,
-  queryParams,
-  metadata,
-) {
+export function getParameterValueFromQueryParams(parameter, queryParams) {
```
Comment (calherries) [inline frontend/src/metabase/parameters/utils/parameter-values.js:4]: `getParameterValueFromQueryParams` also didn't use its `metadata` arg, so I removed it.

File: frontend/src/metabase/parameters/utils/parameter-values.unit.spec.js [hunk 4]
```diff
@@ -8,7 +8,6 @@ describe("parameters/utils/parameter-values", () => {
   let field2;
   let field3;
   let field4;
-  let metadata;
```
Comment (calherries) [inline frontend/src/metabase/parameters/utils/parameter-values.unit.spec.js:11]: All the changes in this file are just me removing `metadata` args

File: frontend/src/metabase/query_builder/actions/querying.js [hunk 5]
```diff
@@ -176,7 +177,7 @@ export const queryCompleted = (question, queryResults) => {
   return async (dispatch, getState) => {
     const [{ data }] = queryResults;
     const [{ data: prevData }] = getQueryResults(getState()) || [{}];
-    const originalQuestion = getOriginalQuestion(getState());
+    const originalQuestion = getOriginalQuestionWithParameterValues(getState());
```
Comment (calherries) [inline frontend/src/metabase/query_builder/actions/querying.js:180]: This is the critical line of the fix. Before, `originalQuestion` wouldn't have `parameterValues` extracted from the card's template tag defaults, but the `question` argument to `queryCompleted` did. So on line 183 below, `isDirty` would be `true` even if the query hadn't changed at all, and `maybeResetDisplay` would be called on line 194. After the change, originalQuestion includes the `parameterValues` from the card's template tag defaults, so `isDirty` would be false. Then `maybeResetDisplay` doesn't get called.

**Review by ranquild — APPROVED** [review]

**Pull request merged**